lib/pattern.c

```diff
@@ -381,6 +381,19 @@ int parse_and_fill_pattern(const char *in, unsigned int in_len,
 	return total;
 }
 
+int parse_and_fill_pattern_alloc(const char *in, unsigned int in_len,
```

Comment:
Maybe add documentation for this function? All other functions in this file have it.